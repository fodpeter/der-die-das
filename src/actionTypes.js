export const LOADING_STATE = {
  LOADING: "LOADING",
  SUCCESS: "SUCCESS",
  FAILED: "FAILED"
};

export const WORDS = {
  LOAD_REQUEST: "LOAD_REQUEST",
  LOAD_SUCCESS: "LOAD_SUCCESS",
  LOAD_FAILED: "LOAD_FAILED"
};
