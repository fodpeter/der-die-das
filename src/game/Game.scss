.game > * {
  margin: 1em;
}
