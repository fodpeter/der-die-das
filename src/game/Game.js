import React from "react";
import PropTypes from "prop-types";
import Label from "react-bootstrap/lib/Label";
import Button from "react-bootstrap/lib/Button";
import { LOADING_STATE } from "../actionTypes";
import "./Game.css";

const Game = ({ loadingState, startGame }) => (
  <div className="game">
    {loadingState === LOADING_STATE.LOADING && (
      <Label bsStyle="info">Loading...</Label>
    )}
    {loadingState === LOADING_STATE.FAILED && (
      <Label bsStyle="danger">Loading failed</Label>
    )}
    <div>
      <Button
        block
        bsSize="large"
        onClick={startGame}
        disabled={loadingState !== LOADING_STATE.READY}
      >
        Start
      </Button>
    </div>
    <a
      href="http://www.passion4teq.com/articles/der-die-das-genus-regeln/"
      target="_blank"
      rel="noopener noreferrer"
    >
      Der die das article rules
    </a>
  </div>
);

Game.propTypes = {
  loadingState: PropTypes.oneOf(Object.values(LOADING_STATE)),
  startGame: PropTypes.func.isRequired
};

export default Game;
