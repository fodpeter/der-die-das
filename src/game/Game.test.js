import React from "react";
import { shallow } from "enzyme";
import { LOADING_STATE } from "../actionTypes";
import Game from "./Game";
import Button from "react-bootstrap/lib/Button";

describe("Game", () => {
  test("renders without crashing", () => {
    const component = shallow(<Game loadingState={LOADING_STATE.READY} />);
    expect(component).toMatchSnapshot();
  });

  test("renders disabled button while loading", () => {
    const component = shallow(<Game loadingState={LOADING_STATE.LOADING} />);
    expect(component.find(Button).prop("disabled")).toBe(true);
  });
});
