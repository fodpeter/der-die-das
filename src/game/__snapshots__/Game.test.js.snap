// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Game
    loadingState="READY"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        false,
        <div>
          <Button
                    active={false}
                    block={true}
                    bsClass="btn"
                    bsSize="large"
                    bsStyle="default"
                    disabled={false}
                    onClick={undefined}
          >
                    Start
          </Button>
</div>,
        <a
          href="http://www.passion4teq.com/articles/der-die-das-genus-regeln/"
          rel="noopener noreferrer"
          target="_blank"
>
          Der die das article rules
</a>,
      ],
      "className": "game",
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            active={false}
            block={true}
            bsClass="btn"
            bsSize="large"
            bsStyle="default"
            disabled={false}
            onClick={undefined}
>
            Start
</Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": true,
            "bsClass": "btn",
            "bsSize": "large",
            "bsStyle": "default",
            "children": "Start",
            "disabled": false,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Start",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Der die das article rules",
          "href": "http://www.passion4teq.com/articles/der-die-das-genus-regeln/",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "Der die das article rules",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          <div>
            <Button
                        active={false}
                        block={true}
                        bsClass="btn"
                        bsSize="large"
                        bsStyle="default"
                        disabled={false}
                        onClick={undefined}
            >
                        Start
            </Button>
</div>,
          <a
            href="http://www.passion4teq.com/articles/der-die-das-genus-regeln/"
            rel="noopener noreferrer"
            target="_blank"
>
            Der die das article rules
</a>,
        ],
        "className": "game",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              active={false}
              block={true}
              bsClass="btn"
              bsSize="large"
              bsStyle="default"
              disabled={false}
              onClick={undefined}
>
              Start
</Button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": true,
              "bsClass": "btn",
              "bsSize": "large",
              "bsStyle": "default",
              "children": "Start",
              "disabled": false,
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Start",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Der die das article rules",
            "href": "http://www.passion4teq.com/articles/der-die-das-genus-regeln/",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Der die das article rules",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
